packages/changed-elements-react/src/widgets/comparisonJobWidget/components/VersionCompareSelectModal.tsx

```diff
@@ -161,133 +262,211 @@ export function VersionCompareSelectDialogV2(props: VersionCompareSelectDialogV2
     }
-    return { startedComparison: false };
+    if (!args.getDialogOpen()) {
+      toastComparisonJobProcessing(args.currentVersion, args.targetVersion);
+    }
+    return { startedComparison: false, comparisonJob: comparisonJob };
   } catch (error) {
+    args.removePendingJob(`${args.targetVersion.changesetId}-${args.currentVersion.changesetId}`);
     toastComparisonJobError(args.currentVersion, args.targetVersion);
     throw error;
   }
 };
 
-const pollForComparisonJobTillComplete = async (args: RunStartComparisonV2Args) => {
-  const getComparisonJob = async () => {
-    const job = (await args.comparisonJobClient.getComparisonJob({
-      iTwinId: args.iModelConnection?.iTwinId as string,
-      iModelId: args.iModelConnection?.iModelId as string,
-      jobId: `${args.targetVersion.changesetId}-${args.currentVersion.changesetId}`,
-      headers: {
-        "Content-Type": "application/json",
-      },
-    })).comparisonJob;
-    return job;
-  };
+type PollForInProgressJobsArgs = {
+  iTwinId: string;
+  iModelId: string;
+  namedVersionLoaderState?: NamedVersionLoaderResult;
+  comparisonJobClient: IComparisonJobClient;
+  iModelConnection: IModelConnection;
+  setResult: (result: NamedVersionLoaderResult) => void;
+  removeRunningJob: (jobId: JobId) => void;
+  getRunningJobs: () => JobAndNamedVersions[];
+  getDialogOpen: () => boolean;
+  getIsDisposed: () => boolean;
+  targetVersion?: NamedVersion;
+};
+
+export const pollForInProgressJobs: (args: PollForInProgressJobsArgs) => Promise<void> = async (args: PollForInProgressJobsArgs) => {
+  void pollUntilCurrentRunningJobsCompleteAndToast(args);
+  if (args.namedVersionLoaderState && args.namedVersionLoaderState.namedVersions.entries.length > 0 && args.getDialogOpen() && !args.getIsDisposed())
+    void pollUpdateCurrentEntriesForModal(args);
+};
 
-  let comparisonJob = await getComparisonJob();
+const pollUntilCurrentRunningJobsCompleteAndToast = async (args: PollForInProgressJobsArgs) => {
   let isConnectionClosed = false;
   args.iModelConnection.onClose.addListener(() => { isConnectionClosed = true; });
   const loopDelayInMilliseconds = 5000;
-
-  while (comparisonJob.status !== "Error" && !isConnectionClosed) {
-    await new Promise((resolve) => setTimeout(resolve, loopDelayInMilliseconds)); // run loop every 5 seconds
-    if (VersionCompare.manager?.isComparing) {
-      return;
+  while (shouldProcessRunningJobs({ getDialogOpen: args.getDialogOpen, getRunningJobs: args.getRunningJobs, isConnectionClosed })) {
+    await new Promise((resolve) => setTimeout(resolve, loopDelayInMilliseconds));
+    for (const runningJob of args.getRunningJobs()) {
+      const completedJob = await args.comparisonJobClient.getComparisonJob({
+        iTwinId: args.iTwinId,
+        iModelId: args.iModelId,
+        jobId: runningJob?.comparisonJob?.comparisonJob.jobId as string,
+      });
+      if (completedJob.comparisonJob.status === "Error") {
```

Comment:
When a network error occurs, an exception will be thrown and the job will never be removed from running list.